cp22/hpi-fun2/newsort/executables/generator.py

```diff
@@ -0,0 +1,55 @@
+import random
+from pathlib import Path
+
+random.seed(-6653597241841577263)
+
+
+def write_testcase(name, desc, numbers, newNumberOrdering):
+    Path(name + '.in').write_text(str(len(numbers)) + '\n' + ' '.join(map(str, numbers)) + '\n' + str(newNumberOrdering))
+    Path(name + '.desc').write_text(desc + '\n')
+
+
+write_testcase('sample1', 'Sample 1 easy', [1, 3, 5, 8], '0123456789')
+write_testcase('sample2', 'Sample 2 easy', [92, 53, 10], 9876543210)
+write_testcase('sample3', 'Sample 3', [47, 40, 99], 2164783059)
+write_testcase('sample4', 'Sample 4', [0, 40, 99], '0164783259')
+write_testcase('wrong_order', 'wrong order', [1, 2], 1234567890)
+write_testcase('same_values', 'numbers are the same', [1, 1], 1234567890)
+write_testcase('different_lengths', 'values have different lengths', [1, 11111], 1234567890)
+write_testcase('different_lengths2', 'values have different lengths', [51111, 123333, 15666666], 1234567890)
```

Comment:
In this case, there are leading zeroes (ones in this case). Is this intended? This is actually not correctly handled by your solution, which doesn't consider leading zeros.

Reply:
I have fixed it by saying that no leading zeros (or whatever its representation is) exist - I filtered them out in the testcases